cJSON.h

```diff
@@ -81,84 +81,113 @@ typedef struct cJSON_Hooks
       void (*free_fn)(void *ptr);
 } cJSON_Hooks;
 
+#if !defined(__WINDOWS__) && (defined(WIN32) || defined(WIN64) || defined(_MSC_VER) || defined(_WIN32))
+#define __WINDOWS__
+#endif
+#ifdef __WINDOWS__
+#if defined(CJSON_DECLARE_STATIC)
+#define CJSON_DECLARE(type)			type __stdcall
+#define CJSON_DECLARE_NONSTD(type)	type __cdecl
+#define CJSON_DECLARE_DATA
+#elif defined(CJSON_EXPORTS)
+#define CJSON_DECLARE(type)			__declspec(dllexport) type __stdcall
+#define CJSON_DECLARE_NONSTD(type)	__declspec(dllexport) type __cdecl
+#define CJSON_DECLARE_DATA			__declspec(dllexport)
+#else
+#define CJSON_DECLARE(type)			__declspec(dllimport) type __stdcall
+#define CJSON_DECLARE_NONSTD(type)	__declspec(dllimport) type __cdecl
+#define CJSON_DECLARE_DATA			__declspec(dllimport)
+#endif
+#else /* !WIN32 */
+#if (defined(__GNUC__) || defined(__SUNPRO_CC) || defined (__SUNPRO_C)) && defined(CJSON_API_VISIBILITY)
+#define CJSON_DECLARE(type)		    __attribute__((visibility("default"))) type
```

Comment:
Does it make sense to declare the `"default"` visibility? If not, please remove.